src/pages/api/hubspot/subscription/index.ts

```diff
@@ -24,22 +19,40 @@ export const getisSubscribed = (
   return isSubscribed;
 };
 
-export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse,
-) {
-  const { email, subscriptionName } = subscriptionRequestSchema.parse(req.body);
-  try {
-    const apiResponse =
-      await hubspot.communicationPreferences.statusApi.getEmailStatus(email);
-
-    const isSubscribed = getisSubscribed(
-      apiResponse.subscriptionStatuses,
-      subscriptionName,
-    );
-
-    return res.status(200).json(isSubscribed);
-  } catch (error) {
-    return res.status(500).json({ error: JSON.stringify(error) });
-  }
+export function createHandler(hubspot: HubspotClient) {
+  return async function handler(req: NextApiRequest, res: NextApiResponse) {
+    const auth = getAuth(req);
+
+    if (!auth.userId) {
+      return res.status(401).send("Unauthorized");
+    }
+
+    const email = (await clerkClient().users.getUser(auth.userId))
```

Comment:
This restricts the endpoint to only enable querying for the status of the authenticated user's email address.